src/ompl/base/OptimizationObjective.h

```diff
@@ -141,8 +144,12 @@ namespace ompl
             /** \brief Returns this objective's SpaceInformation. Needed for operators in MultiOptimizationObjective */
             const SpaceInformationPtr& getSpaceInformation() const;
 
+            /** \brief Allocate a heuristic-sampling state generator for this cost function, defaults to a basic rejection sampling scheme when the derived class does not provide a better method.*/
+            virtual InformedStateSamplerPtr allocInformedStateSampler(const StateSpace* space, const ProblemDefinitionPtr probDefn, const Cost* bestCost) const;
+
```

Comment:
I see this may be tempting, but the sample depends on the state space as much as it does on the optimization objective, so to me it seems better do simply not have this function. We can have the constructor live separately. An allocator function is not really needed. We could set an allocator for this sampler in SpaceInformation, if we so desired.